utils.py

```diff
@@ -131,37 +131,35 @@ def step(self, *args):
         """ Step response """
         return signal.lti(self.numerator, self.denominator).step(*args)
 
-    def simplify(self, dec=5):
+    def simplify(self, dec=3):
 
         # Polynomial simplification
-        g = polygcd(self.numerator, self.denominator)
-        self.numerator, remainder = self.numerator/g
-        assert numpy.allclose(remainder.coeffs, 0, atol=1e-6), "Error in simplifying rational, remainder=\n{}".format(remainder)
-        self.denominator, remainder = self.denominator/g
-        assert numpy.allclose(remainder.coeffs, 0, atol=1e-6), "Error in simplifying rational, remainder=\n{}".format(remainder)
-
-        # Round numerator and denominator for coefficient simplification
-        self.numerator = numpy.poly1d(numpy.round(self.numerator, dec))
-        self.denominator = numpy.poly1d(numpy.round(self.denominator, dec))
-
-        # Determine most digits in numerator & denominator
-        num_dec = 0
-        den_dec = 0
-        for i in range(len(self.numerator.coeffs)):
-            num_dec = max(num_dec, decimals(self.numerator.coeffs[i]))
-        for j in range(len(self.denominator.coeffs)):
-            den_dec = max(den_dec, decimals(self.denominator.coeffs[j]))
-
-        # Convert coefficients to integers
-        self.numerator = self.numerator*10**(max(num_dec, den_dec))
-        self.denominator = self.denominator*10**(max(num_dec, den_dec))
-
-        # Decimal-less representation of coefficients
-        num_gcd = gcd(self.numerator.coeffs)
-        den_gcd = gcd(self.denominator.coeffs)
-        tf_gcd = gcd([num_gcd, den_gcd])
-        self.numerator = self.numerator/tf_gcd
-        self.denominator = self.denominator/tf_gcd
+        k = self.numerator[self.numerator.order]/self.denominator[self.denominator.order]
+        ps = self.poles().tolist()
+        zs = self.zeros().tolist()
+        ps_to_canc_ind = []    #contains index of poles to be cancelled
+        zs_to_canc_ind = []    #contains index of poles to be cancelled
+        
+        for i in range(0, len(zs)):
```

Comment:
We can avoid checking all possibilities by sorting on the real component, because the imaginary parts always come in pairs, the real part gives us a good ordering.